internal/cephfs/controllerserver.go

```diff
@@ -974,6 +974,62 @@ func (cs *ControllerServer) validateSnapshotReq(ctx context.Context, req *csi.Cr
 	return nil
 }
 
+func (cs *ControllerServer) ListSnapshots(
+	ctx context.Context,
+	req *csi.ListSnapshotsRequest,
+) (*csi.ListSnapshotsResponse, error) {
+	if err := cs.Driver.ValidateControllerServiceRequest(
+		csi.ControllerServiceCapability_RPC_LIST_SNAPSHOTS); err != nil {
+		log.ErrorLog(ctx, "invalid list snapshot req: %v", protosanitizer.StripSecrets(req))
+
+		return nil, err
+	}
+
+	cr, err := util.NewAdminCredentials(req.GetSecrets())
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	defer cr.DeleteCredentials()
+
+	snapshotID := req.GetSnapshotId()
+	if snapshotID == "" {
+		return nil, status.Error(codes.InvalidArgument, "snapshot ID cannot be empty")
+	}
```

Comment:
Technically, we can't error out here because the field is `optional`.
So, if the snapshotID is missing, the csi-driver should be listing all the available snapshots in the backend.
Ref: https://github.com/container-storage-interface/spec/blob/master/lib/go/csi/csi.pb.go#L3527

Reply:
That is a problem for us as without the snapshot ID we won't really know where to list snapshots from :(

Reply:
And, that's the reason I think either we shouldn't implement this at all (since it's an optional PRC) or don't error out if the id is not present because it goes against the RPC definition of the field being optional.